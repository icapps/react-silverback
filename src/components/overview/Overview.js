import React from 'react';
import PropTypes from 'prop-types';
import { Pagination, Table, CreateModal, Filter, Alert } from '../index';
import { strings } from '../../utils';
import constants from '../../redux/users/constants';
import './overview.css';

const SORT_DESC = 'desc';

class Overview extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 0,
      limit: 25,
      deletedItem: props.deletedItem ? props.deletedItem : '',
      actionText: '',
      actionClass: '',
      filterValue: '',
    };
    this.timer = null;
  }

  componentDidMount() {
    this.props.get(this.state.page, this.state.limit);
  }

  componentWillUnmount() {
    this.props.resetDeletedItem();
  }

  remove = async (item, deletedItem) => {
    if (this.props.removeItem) {
      const result = await this.props.removeItem(item);
      if (result.action && result.action.type === constants.REMOVE_USER_FULFILLED) {
        this.scrollTop();
        this.setState({ deletedItem });
        this.props.get(this.state.page * this.state.limit, this.state.limit, this.props.sortField, this.props.sortOrder);
      }
    }
  }

  setActions = actions => {
    return actions.map(action => {
      return {
        ...action, handleAction: async item => {
          const actionResult = await action.handleAction(item.id);
          if (actionResult.action && actionResult.action.type.includes('FULFILLED')) {
            this.scrollTop();
            this.setState({ actionClass: action.actionClassName, actionText: strings.formatString(action.successMessage, { item: <strong>{item[this.props.deleteIdentifier]}</strong> }) });
          }
          await this.sortItems(this.props.sortField, this.props.sortOrder);
        },
      };
    });
  }

  showDetailScreen = id => {
    this.props.history.push(`${window.location.pathname}/${id}`, id);
  };

  sortItems = (sortField, sortOrder) => {
    this.props.get(this.state.page * this.state.limit, this.state.limit, sortField, sortOrder, this.state.filterValue);
  }

  handlePagination = (page, limit) => {
    this.props.get(page * limit, limit, this.props.sortField, this.props.sortOrder);
    this.setState({ page, limit });
  };

  handleFilter = filterValue => {
    clearTimeout(this.timer);
    this.timer = setTimeout(() => {
      this.props.get(this.state.page * this.state.limit, this.state.limit, this.props.sortField, this.props.sortOrder, filterValue);
      this.setState({ filterValue });
    }, 700);
  }

  scrollTop = () => {
    window.scrollTo(0, 0);
  }

  getAlerts = () => {
    const alerts = [];

    if (this.props.isError) {
      alerts.push(<Alert className={'danger'} text={this.state.actionText} key={new Date()} clearAlerts={this.clearAlerts} />);
    }
    if (this.state.actionText !== '') {
      alerts.push(<Alert className={this.state.actionClass} text={this.state.actionText[0][0].props.children + this.state.actionText[1]} key={'t'} clearAlerts={this.clearAlerts} />);
    }
    if (this.state.deletedItem !== '') {
      alerts.push(<Alert className={'success'} text={strings.formatString(strings.DELETED_ITEM, { item: <strong>{this.state.deletedItem}</strong> })} key={new Date()} clearAlerts={this.clearAlerts} />);
    }
    return alerts;
  }

  clearAlerts = () => {
    this.setState({
      actionText: '',
      deletedItem: '',
    });
  }

  render() {
    const { props, state } = this;
    return (
      <main className="overview col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3">
        <div className="container">
          {this.getAlerts()}
          <h2>
            {props.title}
            {props.sortField && <span className="sort-label">{`${strings.SORTED_BY} ${props.sortField} (${props.sortOrder === SORT_DESC ? strings.DESCENDING : strings.ASCENDING})`}</span>}
          </h2>
          <div className="overview-settings">
            <Filter handleFilter={this.handleFilter} />
            {props.create && <CreateModal
              primaryButtonText={`${strings.CREATE} ${props.keyword.toLowerCase()}`}
              title={`${strings.CREATE} ${props.keyword.toLowerCase()}`}
              createParameters={this.props.createParameters}
              create={this.props.create}
              isError={props.isCreateError}
              isPending={props.isCreatePending}
              errorMessage={props.errorMessage}
            />}
          </div>
          {props.listItems.length > 0 ? (
            <React.Fragment>
              <Pagination totalCount={props.paginationTotalCount} handleClick={this.handlePagination} activePage={state.page} pageLimit={state.limit} />
              <Table
                keys={props.keys}
                listItems={props.listItems}
                dateFormat={props.dateFormat}
                handleRowClick={this.showDetailScreen}
                handleRemoveItem={this.props.removeItem && this.remove}
                handleSort={this.sortItems}
                deleteIdentifier={props.deleteIdentifier}
                actions={this.setActions(this.props.actions)}
                sortOrder={props.sortOrder}
                sortField={props.sortField}
              />
              <Pagination totalCount={props.paginationTotalCount} handleClick={this.handlePagination} activePage={state.page} pageLimit={state.limit} />
            </React.Fragment>
          ) : <div className="jumbotron" role="alert"><span className="empty-overview">{strings.formatString(strings.NO_RESULTS_FOUND, { result: props.title })}</span></div>
          }
        </div>
      </main>
    );
  }
};

Overview.propTypes = {
  title: PropTypes.string.isRequired,
  keyword: PropTypes.string,
  keys: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    value: PropTypes.string,
    isSortable: PropTypes.bool,
  })).isRequired,
  listItems: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool, PropTypes.object])).isRequired,
  get: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  paginationTotalCount: PropTypes.number.isRequired,
  create: PropTypes.func,
  createParameters: PropTypes.array,
  dateFormat: PropTypes.string,
  removeItem: PropTypes.func,
  deleteIdentifier: PropTypes.string,
  isError: PropTypes.bool.isRequired,
  isCreatePending: PropTypes.bool,
  isCreateError: PropTypes.bool,
  errorMessage: PropTypes.string.isRequired,
  actions: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    label: PropTypes.string,
    handleAction: PropTypes.func,
    primaryButtonText: PropTypes.string,
  })),
  deletedItem: PropTypes.string,
  resetDeletedItem: PropTypes.func,
  sortOrder: PropTypes.string,
  sortField: PropTypes.string,
};

Overview.defaultProps = {
  create: null,
  createParameters: [],
  dateFormat: null,
  removeItem: null,
  deleteIdentifier: '',
  deletedItem: '',
  resetDeletedItem: () => { },
  keyword: '',
  isCreatePending: false,
  isCreateError: false,
  actions: [],
  sortOrder: '',
  sortField: '',
};

export default Overview;
