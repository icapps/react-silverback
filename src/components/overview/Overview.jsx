import React from 'react';
import PropTypes from 'prop-types';
import { Table } from '../index';
import './overview.css';

const Overview = props => (
  <main className="col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3">
    <h2>{props.title}</h2>
    {props.listItems.length > 0 ? (
      <Table
        keys={props.keys}
        listItems={props.listItems}
        dateFormat={props.dateFormat}
        handleRowClick={() => { }}
        handleRemoveItem={() => { }}
        handleSort={() => { }}
      />
    ) : <div className="jumbotron" role="alert"><span className="empty-overview">No {props.title} found ...</span></div>
    }
  </main>
);

Overview.propTypes = {
  title: PropTypes.string.isRequired,
  keys: PropTypes.array.isRequired,
  listItems: PropTypes.array.isRequired,
  dateFormat: PropTypes.string,
};

export default Overview;
