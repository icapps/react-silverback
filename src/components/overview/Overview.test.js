import React from 'react';
import { shallow, configure } from 'enzyme';
import Overview from './Overview';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('Overview Component', () => {
  it('should render an Overview component', () => {
    const wrapper = shallow(<Overview
      title="DataTypes"
      keys={[{ id: 'id', value: 'id', isSortable: true }, { id: 'type', value: 'type', isSortable: true }, { id: 'price', value: 'price', isSortable: false }]}
      listItems={[{ id: '1', type: 'B', price: 88.9 }, { id: '2', type: 'C', price: 45.2 }, { id: '1', type: 'D', price: 56.5 }]}
      removeItem={() => { }}
      get={() => { }}
      history={{}}
      paginationTotalCount={100}
      createParameters={[
        { id: 'key1', label: "test", type: "text" },
        { id: 'key2', label: "test2", type: "text" },
        { id: 'key3', label: "test3", type: "number" },
        { id: 'key4', label: "test4", type: "boolean" },
      ]}
      create={() => { }}
      isError={false}
      errorMessage="test"
      setMessage={() => { }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render an Overview component with custom date format', () => {
    const wrapper = shallow(<Overview
      title="DataTypes"
      keys={[{ id: 'id', value: 'id', isSortable: true }, { id: 'type', value: 'type', isSortable: true }, { id: 'date', value: 'date', isSortable: false }]}
      listItems={[{ id: '1', type: 'B', date: new Date(Date.UTC(1995, 4, 23)) }, { id: '2', type: 'C', date: new Date(Date.UTC(1995, 4, 23)) }, { id: '1', type: 'D', date: new Date(Date.UTC(1995, 4, 23)) }]}
      removeItem={() => { }}
      get={() => { }}
      history={{}}
      paginationTotalCount={100}
      createParameters={[
        { id: 'key1', label: "test", type: "text" },
        { id: 'key2', label: "test2", type: "text" },
        { id: 'key3', label: "test3", type: "number" },
        { id: 'key4', label: "test4", type: "boolean" },
      ]}
      create={() => { }}
      isError={false}
      errorMessage="test"
      setMessage={() => { }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render an Overview component with no listItems', () => {
    const wrapper = shallow(<Overview
      title="DataTypes"
      keys={[{ id: 'id', value: 'id', isSortable: true }, { id: 'type', value: 'type', isSortable: true }, { id: 'price', value: 'price', isSortable: false }]}
      listItems={[]}
      removeItem={() => { }}
      get={() => { }}
      history={{}}
      paginationTotalCount={100}
      createParameters={[
        { id: 'key1', label: "test", type: "text" },
        { id: 'key2', label: "test2", type: "text" },
        { id: 'key3', label: "test3", type: "number" },
        { id: 'key4', label: "test4", type: "boolean" },
      ]}
      create={() => { }}
      isError={false}
      errorMessage="test"
      setMessage={() => { }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
