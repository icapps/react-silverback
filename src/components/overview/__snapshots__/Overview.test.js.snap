// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Component should render a Overview component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Overview
    keys={
        Array [
            "id",
            "type",
            "price",
          ]
    }
    listItems={
        Array [
            Object {
              "id": "1",
              "price": 88.9,
              "type": "B",
            },
            Object {
              "id": "2",
              "price": 45.2,
              "type": "C",
            },
            Object {
              "id": "1",
              "price": 56.5,
              "type": "D",
            },
          ]
    }
    title="DataTypes"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          DataTypes
</h2>,
        <Table
          dateFormat="MM/DD/YYYY"
          handleRemoveItem={[Function]}
          handleRowClick={[Function]}
          handleSort={[Function]}
          keys={
                    Array [
                              "id",
                              "type",
                              "price",
                            ]
          }
          listItems={
                    Array [
                              Object {
                                "id": "1",
                                "price": 88.9,
                                "type": "B",
                              },
                              Object {
                                "id": "2",
                                "price": 45.2,
                                "type": "C",
                              },
                              Object {
                                "id": "1",
                                "price": 56.5,
                                "type": "D",
                              },
                            ]
          }
/>,
      ],
      "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DataTypes",
        },
        "ref": null,
        "rendered": "DataTypes",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dateFormat": "MM/DD/YYYY",
          "handleRemoveItem": [Function],
          "handleRowClick": [Function],
          "handleSort": [Function],
          "keys": Array [
            "id",
            "type",
            "price",
          ],
          "listItems": Array [
            Object {
              "id": "1",
              "price": 88.9,
              "type": "B",
            },
            Object {
              "id": "2",
              "price": 45.2,
              "type": "C",
            },
            Object {
              "id": "1",
              "price": 56.5,
              "type": "D",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            DataTypes
</h2>,
          <Table
            dateFormat="MM/DD/YYYY"
            handleRemoveItem={[Function]}
            handleRowClick={[Function]}
            handleSort={[Function]}
            keys={
                        Array [
                                    "id",
                                    "type",
                                    "price",
                                  ]
            }
            listItems={
                        Array [
                                    Object {
                                      "id": "1",
                                      "price": 88.9,
                                      "type": "B",
                                    },
                                    Object {
                                      "id": "2",
                                      "price": 45.2,
                                      "type": "C",
                                    },
                                    Object {
                                      "id": "1",
                                      "price": 56.5,
                                      "type": "D",
                                    },
                                  ]
            }
/>,
        ],
        "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DataTypes",
          },
          "ref": null,
          "rendered": "DataTypes",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dateFormat": "MM/DD/YYYY",
            "handleRemoveItem": [Function],
            "handleRowClick": [Function],
            "handleSort": [Function],
            "keys": Array [
              "id",
              "type",
              "price",
            ],
            "listItems": Array [
              Object {
                "id": "1",
                "price": 88.9,
                "type": "B",
              },
              Object {
                "id": "2",
                "price": 45.2,
                "type": "C",
              },
              Object {
                "id": "1",
                "price": 56.5,
                "type": "D",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Overview Component should render a Overview component with custom date format 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Overview
    keys={
        Array [
            "id",
            "type",
            "date",
          ]
    }
    listItems={
        Array [
            Object {
              "date": 1995-05-23T00:00:00.000Z,
              "id": "1",
              "type": "B",
            },
            Object {
              "date": 1995-05-23T00:00:00.000Z,
              "id": "2",
              "type": "C",
            },
            Object {
              "date": 1995-05-23T00:00:00.000Z,
              "id": "1",
              "type": "D",
            },
          ]
    }
    title="DataTypes"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          DataTypes
</h2>,
        <Table
          dateFormat="MM/DD/YYYY"
          handleRemoveItem={[Function]}
          handleRowClick={[Function]}
          handleSort={[Function]}
          keys={
                    Array [
                              "id",
                              "type",
                              "date",
                            ]
          }
          listItems={
                    Array [
                              Object {
                                "date": 1995-05-23T00:00:00.000Z,
                                "id": "1",
                                "type": "B",
                              },
                              Object {
                                "date": 1995-05-23T00:00:00.000Z,
                                "id": "2",
                                "type": "C",
                              },
                              Object {
                                "date": 1995-05-23T00:00:00.000Z,
                                "id": "1",
                                "type": "D",
                              },
                            ]
          }
/>,
      ],
      "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DataTypes",
        },
        "ref": null,
        "rendered": "DataTypes",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dateFormat": "MM/DD/YYYY",
          "handleRemoveItem": [Function],
          "handleRowClick": [Function],
          "handleSort": [Function],
          "keys": Array [
            "id",
            "type",
            "date",
          ],
          "listItems": Array [
            Object {
              "date": 1995-05-23T00:00:00.000Z,
              "id": "1",
              "type": "B",
            },
            Object {
              "date": 1995-05-23T00:00:00.000Z,
              "id": "2",
              "type": "C",
            },
            Object {
              "date": 1995-05-23T00:00:00.000Z,
              "id": "1",
              "type": "D",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            DataTypes
</h2>,
          <Table
            dateFormat="MM/DD/YYYY"
            handleRemoveItem={[Function]}
            handleRowClick={[Function]}
            handleSort={[Function]}
            keys={
                        Array [
                                    "id",
                                    "type",
                                    "date",
                                  ]
            }
            listItems={
                        Array [
                                    Object {
                                      "date": 1995-05-23T00:00:00.000Z,
                                      "id": "1",
                                      "type": "B",
                                    },
                                    Object {
                                      "date": 1995-05-23T00:00:00.000Z,
                                      "id": "2",
                                      "type": "C",
                                    },
                                    Object {
                                      "date": 1995-05-23T00:00:00.000Z,
                                      "id": "1",
                                      "type": "D",
                                    },
                                  ]
            }
/>,
        ],
        "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DataTypes",
          },
          "ref": null,
          "rendered": "DataTypes",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dateFormat": "MM/DD/YYYY",
            "handleRemoveItem": [Function],
            "handleRowClick": [Function],
            "handleSort": [Function],
            "keys": Array [
              "id",
              "type",
              "date",
            ],
            "listItems": Array [
              Object {
                "date": 1995-05-23T00:00:00.000Z,
                "id": "1",
                "type": "B",
              },
              Object {
                "date": 1995-05-23T00:00:00.000Z,
                "id": "2",
                "type": "C",
              },
              Object {
                "date": 1995-05-23T00:00:00.000Z,
                "id": "1",
                "type": "D",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Overview Component should render a Overview component with no listItems 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Overview
    keys={
        Array [
            "id",
            "type",
            "price",
          ]
    }
    listItems={Array []}
    title="DataTypes"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          DataTypes
</h2>,
        <div
          className="jumbotron"
          role="alert"
>
          <span
                    className="empty-overview"
          >
                    No 
                    DataTypes
                     found ...
          </span>
</div>,
      ],
      "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DataTypes",
        },
        "ref": null,
        "rendered": "DataTypes",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="empty-overview"
>
            No 
            DataTypes
             found ...
</span>,
          "className": "jumbotron",
          "role": "alert",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "No ",
              "DataTypes",
              " found ...",
            ],
            "className": "empty-overview",
          },
          "ref": null,
          "rendered": Array [
            "No ",
            "DataTypes",
            " found ...",
          ],
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            DataTypes
</h2>,
          <div
            className="jumbotron"
            role="alert"
>
            <span
                        className="empty-overview"
            >
                        No 
                        DataTypes
                         found ...
            </span>
</div>,
        ],
        "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DataTypes",
          },
          "ref": null,
          "rendered": "DataTypes",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="empty-overview"
>
              No 
              DataTypes
               found ...
</span>,
            "className": "jumbotron",
            "role": "alert",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "No ",
                "DataTypes",
                " found ...",
              ],
              "className": "empty-overview",
            },
            "ref": null,
            "rendered": Array [
              "No ",
              "DataTypes",
              " found ...",
            ],
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
